### ISSUE TITLE
require-data-selectors and no-xpath don't support chained calls

### ISSUE THREAD

**Issue by MgmClientGuy0** [cypress-io/eslint-plugin-cypress#291]
All rules should work when chaining multiple calls together. While testing, I found that require-data-selectors and no-xpath are the exception. For example, I believe that the following example should be detected by require-data-selectors.

```js
cy.should("exist").get(".selector");
```

**Comment by MikeMcC399** [comment]
@MgmClientGuy0 

```shell
cy.should("exist").get(".selector");
```

Apart from the rule [require-data-selectors](https://github.com/cypress-io/eslint-plugin-cypress/blob/master/docs/rules/require-data-selectors.md) this is incorrect chaining of [should](https://docs.cypress.io/api/commands/should)

<img width="1012" height="354" alt="Image" src="https://github.com/user-attachments/assets/05298fb9-3500-43c7-9ffe-ba266d5bb042" />

Do you have an example of valid code which should fail the [require-data-selectors](https://github.com/cypress-io/eslint-plugin-cypress/blob/master/docs/rules/require-data-selectors.md) rule but does not fail?

**Comment by MgmClientGuy0** [comment]
@MikeMcC399 Yeah makes sense, but should "no-chained-get" really mark all commands before the .get() as invalid? I was originally thinking to just error on commands that yield a DOM node, since these get ignored by .get() and have no effect on their own. My second example with the .get() chained of the .log() seems valid right?

Regarding the no-xpath, I have no opinion on this. I wanted to fix this just for the sake of consistency (so all rules have the same way of determining if it a Cypress chain), but I will leave it as it is then.
